test/dynamo/test_misc.py

```diff
@@ -7242,45 +7242,78 @@ def fn(target):
         res = opt_func(a)
         self.assertIsInstance(res, torch.Tensor)
 
+    def test_itertools_accumulate_symint_default_sum(self):
+        from torch._dynamo.utils import counters
+
+        counters.clear()
+
+        def fn(x):
+            r = itertools.accumulate([x.size(0), x.size(1)])
+            for i in r:
+                x *= i
+            return x
+
+        x = torch.randn(2, 3)
+        eager = fn(x)
+
+        counter = CompileCounter()
+        compiled_fn = torch._dynamo.optimize(counter)(fn)
+        compiled = compiled_fn(x)
+
+        self.assertEqual(list(eager), list(compiled))
+        self.assertEqual(len(counters["graph_break"]), 0)
+        self.assertEqual(counter.frame_count, 1)
+
     def test_itertools_accumulate_tensors_default_sum(self):
+        from torch._dynamo.utils import counters
+
+        counters.clear()
+
         def fn(a, b, c, d, x):
             l = [a, b, c, d, x]
             for i, t in enumerate(l):
                 l[i] = t * x
-            return itertools.accumulate(l)
+            return list(itertools.accumulate(l))
 
-        t_list = [torch.tensor([i]) for i in range(4)]
-        x = torch.randn([2, 2])
+        t_list = [torch.tensor([i + 1]) for i in range(4)]
+        x = torch.tensor([[1, 2], [3, 4]])
         eager = fn(*t_list, x)
 
         counter = CompileCounter()
         compiled_fn = torch._dynamo.optimize(counter)(fn)
         compiled = compiled_fn(*t_list, x)
 
         self.assertEqual(list(eager), list(compiled))
+        self.assertEqual(len(counters["graph_break"]), 0)
         self.assertEqual(counter.frame_count, 1)
 
     def test_itertools_accumulate_tensors_builtins(self):
+        from torch._dynamo.utils import counters
+
         for builtin_op in [operator.mul, operator.sub, operator.pow]:
+            counters.clear()
 
             def fn(a, b, c, d, x):
                 l = [a, b, c, d, x]
                 for i, t in enumerate(l):
                     l[i] = t * x
-                return itertools.accumulate(l, func=builtin_op)
+                return list(itertools.accumulate(l, builtin_op))
```

Comment:
I don't understand why you are coercing to a list here? It should. be sound to return an accumulate.

Reply:
Hmm, perhaps coercing to list was not strictly necessary. However, I think that due to not checking graph break and using kwarg (which was unsupported) the test mistakenly didn't test anything.

Let me try without list

Reply:
to avoid graph breaks, best to fullgraph=True in the compile call

Reply:
self.assertEqual(counter.frame_count, 1) 

Also means you didn't graph break.

Reply:
> self.assertEqual(counter.frame_count, 1)
> 
> Also means you didn't graph break.

Strangely, this was not what I observed to be the case

I saw from the logs that `graph_breaks [..]` was non-empty, and showed an error in calling accumulate

Reply:
Perhaps it's because calling `accumulate` is the final instruction before ret in the calling frame?

Reply:
Tracking this as a bad practice now: https://github.com/pytorch/pytorch/issues/110730

Reply:
I've updated diagnosis of the root cause of bad tests in PR description.